Update README.md (#14)
It4innovations/permon: README.md

PERMON toolbox
================================

> PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for quadratic programming (QP) based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) and Support Vector Machines (SVM) machine learning methods.

Reviewer: It should be clear that this repository only contains QP stuff and FETI

Author: Would this suffice? https://github.com/permon/permon/pull/14/files#r417501398

Reviewer: It is probably sufficient that it is in the repository description.

I suggest changing the repository description to  "PERMON main package for quadratic programming (PermonQP)".

> PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for quadratic programming (QP) based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) and Support Vector Machines (SVM) machine learning methods.

Reviewer: ```suggestion
PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for the scalable (parallel) solution of  quadratic programming (QP) problems based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) and Support Vector Machines (SVM) machine learning methods.
```

Reviewer: > It also includes

It may seem a bit random that a QP toolbox includes these. Should we somehow clarify that these are QP problems?

Reviewer: > FETI

Shouldn't we stress that the combination with the QP solvers gives the ability to (efficiently) solve contact problems?

Reviewer: > > FETI
> 
> Shouldn't we stress that the combination with the QP solvers gives the ability to (efficiently) solve contact problems?

Also maybe FETI can be subpoint in the PermonQP point

Project homepage: <http://permon.vsb.cz>  

PERMON contains following modules:

> * [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP solvers and (Total) FETI domain decomposition methods

Author: ```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) (this repository) - contains QP solvers and (Total) FETI domain decomposition methods
```

Author: superseded by https://github.com/permon/permon/pull/14#discussion_r419310866

> * [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP solvers and (Total) FETI domain decomposition methods

Reviewer: ```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - solvers and routines for QP problems and (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods
```

The fllop web page should be eventually changed to FETI which should be a subsection of PermonQP.

> * [PermonQP](http://permon.vsb.cz/permonqp.htm) - solvers and routines for QP problems and (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods

Author: ```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - this repository
  - solvers and routines for solution of QP problems
  - constrained (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods - solve contact problems with domain decomposition
```

Author: @jkruzik

Reviewer: We could even drop "this repository".
I would suggest changing "domain decomposition methods - solve contact problems with domain decomposition" to
"domain decomposition methods able to solve contact problems"

Author: > We could even drop "this repository".

I would like to keep it for clarity as long as the PermonQP name is used. But maybe we could abandon this name. It's just PERMON, and PermonSVM is its extension for SVM.

> I would suggest changing "domain decomposition methods - solve contact problems with domain decomposition" to
"domain decomposition methods able to solve contact problems"

Updated suggestion:
```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - this repository
  - solvers and routines for solution of QP problems
  - constrained (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods able to solve contact problems
```

Reviewer: OK, keep it.
> It's just PERMON...

You can suggest it by mail to the others.

> * [PermonSVM](http://permon.vsb.cz/permonsvm.htm) - SVM machine learning implementation based on PermonQP.

Reviewer: There is no link to PermonSVM respository

Author: ```suggestion
* [PermonSVM](http://permon.vsb.cz/permonsvm.htm) [(another repo)](https://github.com/permon/permonsvm) - SVM machine learning implementation based on PermonQP.
```

> * [PermonSVM](http://permon.vsb.cz/permonsvm.htm) - SVM machine learning implementation based on PermonQP.

Reviewer: ```suggestion
* [PermonSVM](http://permon.vsb.cz/permonsvm.htm) - SVM machine learning implementation based on PermonQP - [separate repository](https://github.com/permon/permonsvm)
```

Supported PETSc versions
----------------------------------
PERMON tries to support newest versions of PETSc as soon as possible. The [releases](https://github.com/permon/permon/releases) are tagged with major.minor.sub-minor numbers. The major.minor numbers correspond to the major.minor release numbers of the supported PETSc version.

> 

Reviewer: There should be a new line here.

Author: I wonder why but that's not worth further discussion.

Author: If I try to edit the file, the newline is there. It seems it's just some GitHub noise. Closing.

Reviewer: I think that GitHub shows the new line as missing making the last line not being a POSIX line. Not a big deal in README. It should not happen in a code as some compilers still throw warnings (e.g., CRAY).
